addons/api/addon/serializers/credential-library.js

```diff
@@ -14,16 +17,39 @@ export default class CredentialLibrarySerializer extends ApplicationSerializer {
    * @method serialize
    * @param {Snapshot} snapshot
    */
-  serialize() {
+  serialize(snapshot) {
+    switch (snapshot.record.type) {
+      case TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC:
+        return this.serializeVaultGeneric(...arguments);
+      case TYPE_CREDENTIAL_LIBRARY_VAULT_SSH_CERTIFICATE:
+      default:
+        return super.serialize(...arguments);
+    }
+  }
+
+  serializeAttribute(snapshot, json, key, attribute) {
+    super.serializeAttribute(...arguments);
+    const { options } = attribute;
+    const { type } = snapshot.record;
+
+    // For any attribute that doesn't match its `for`
+    // or isn't undefined, we delete it from the json
+    if (
+      options.isNestedAttribute &&
+      options.for !== type &&
+      options.for !== undefined
+    ) {
+      delete json.attributes[key];
+    }
```

Comment:
Tried to reframe this in the positive.  IMHO conditionals stated
in the positive are easier to understand.  I'm also wondering
if it matters if an attribute is nested, shouldn't it match the
record type as long as `for` is specified?  And does it matter
if options.for is strictly not undefined or simply that it has
a value?  We can always get strict about typing later on when
we introduce formal resource types.
```suggestion
    // If an attribute has a `for` option, it must match the
    // record's `type`, else the attribute is excluded
    // from serialization.
    if (options?.for && options.for !== type) {
      delete json.attributes[key];
    }
```

Reply:
Yeah it shouldn't matter whether it's a nested attribute, I didn't have it originally but don't recall why I decided to add it, I think I might have actually just been testing, I'll change this. 

I do agree I think this is easier to read, I'll update it